addons/pos_self_order/controllers/manifest_controller.py

```diff
@@ -0,0 +1,73 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+# import base64
+import json
+
+from odoo import http
+from odoo.http import request
+from odoo.tools import ustr, file_open
+
+
+class PosManifest(http.Controller):
+
+    @http.route('/pos-self/manifest', type='http', auth='public', methods=['GET'])
+    def posmanifest(self):
+        """ Returns a WebManifest describing the metadata associated with a web application.
+        Using this metadata, user agents can provide developers with means to create user
+        experiences that are more comparable to that of a native application.
+        """
+        company_name = request.env.company.name
+        manifest = {
+            'name': '%s' % company_name,
+            'short_name': '%s' % company_name,
+            'scope': '/pos-self',
+            'start_url': '/pos-self',
+            'display': 'standalone',
+            'background_color': '#714B67',
+            'theme_color': '#714B67',
+            'orientation': 'portrait-primary'
+        }
+        icon_sizes = ['192', '512']
+        manifest['icons'] = [{
+            'src': self._icon_path(size),
+            'sizes': size + 'x' + size,
+            'type': 'image/png',
+        } for size in icon_sizes]
+        body = json.dumps(manifest, default=ustr)
+        response = request.make_response(body, [
+            ('Content-Type', 'application/manifest+json'),
+        ])
+        return response
+
+    @http.route('/pos-self/service-worker.js', type='http', auth='public', methods=['GET'])
+    def service_worker(self):
+        response = request.make_response(
+            self._get_service_worker_content(),
+            [
+                ('Content-Type', 'text/javascript'),
+                ('Service-Worker-Allowed', '/pos-self'),
+            ]
+        )
+        return response
+
+    def _get_service_worker_content(self):
+        """ Returns a ServiceWorker javascript file scoped for the backend (aka. '/pos-self')
+        """
+        with file_open('pos_self_order/static/src/service_worker.js') as f:
+            body = f.read()
+            return body
+
+    def _icon_path(self, size):
+        company = request.env.company
+        company_id = request.env.company.id
+        if company.logo != company._get_logo():
+            icon_src = '/web/image?model=res.company&id=%s&width=%s' % (company_id, size)
+        else:
+            icon_src = '/web/static/img/odoo-icon-%sx%s.png' % (size, size)
+        return icon_src
+
+    @http.route('/pos-self/offline', type='http', auth='public', methods=['GET'])
+    def offline(self):
+        """ Returns the offline page delivered by the service worker """
+        return request.render('web.webclient_offline', {
+            # 'odoo_icon': base64.b64encode(file_open(self._icon_path('192'), 'rb').read())
+        })
```

Comment:
can you just use the `/web/offline` route instead of creating a custom one for pos?

Reply:
I could but I want to use my icon